src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -173,6 +173,14 @@ describe('Decimal', () => {
       );
     });
 
+    it('is not triggered when Decimal has readOnly prop', () => {
+      render({ readOnly: true });
+      paste({ key: 'invalid' }, '0.|00');
+      expect(wrapper.current.find(StyledWiggle).prop('isAnimating')).toBe(false);
+      press({ key: '-' }, '0.|00');
+      expect(wrapper.current.find(StyledWiggle).prop('isAnimating')).toBe(false);
```

Comment:
Without triggering `onAnimationEnd` this assertion could never be `true`. Please, can you look to see if we can make these assertions without relying on the `isAnimating` prop?